### DISCUSSION TITLE
Dealing with solutions in assignment repos under new student repo creation mechanism

### DISCUSSION THREAD

**Discussion by john-french** [community/community#110536] [GitHub Education] [answered]
### Hi everyone,

Hi,
I'm trying to work out what the optimal workflow will be under the new student assignment repo mechanism.

I have template repos, with long commit histories that include solutions, and solution branches. My understanding is that if I try to use these repos as the starter code repos for my assignments, students will get all of my commit history and the solution branch along with their fork.

My starter code repos live in a master org. Each year that year's class gets a new org. If I just use the starter code template repos in the master org to create new repos in the class org, then these should be fine to fork for student repos (the solution branch and commit history won't be brought over from the master org template).

Am I missing anything here? By the way, as other users have reported, the problem this feature is coming in to solve doesn't really exist for me, trying to ensure I don't get caught out by the change is a much bigger headache.

Thanks,
John.

**Comment by RyanHecht** [comment, marked as answer]
Hi John,

Doesn't look to me like you're missing anything! That workflow sounds perfect to ensure that the old solutions and branches don't get sent out to students -- when you create a repository from a template, that should "squash" all the history for the resulting repo, and then using _that_ for starter code will ensure students can't look at the solutions!

Thanks for the feedback that this new mode of assignment repository creation adds headache -- the team is keeping this in mind and considering quality of life improvements before General Availability to make this transition as easy as possible for all teachers!

**Reply by john-french** [reply]
Ok, thanks for the clarification @RyanHecht .
